policyengine_core/parameters/parameter_node.py

```diff
@@ -274,3 +277,27 @@ def get_child(self, path: str) -> "ParameterNode":
                     f"Could not find the parameter (failed at {name})."
                 )
         return node
+
+    def get_attr_dict(self) -> dict:
+        data = self.__dict__.copy()
```

Comment:
Just one minor suggestion: in this function, could we process every attr that isn't in `child_dict`, then process all the children? This makes keys at the end of the alphabet (like "tracer") more easily readable.